tgui/packages/tgui/interfaces/KeyComboModal.tsx

```diff
@@ -0,0 +1,148 @@
+import { KEY } from 'common/keys';
+
+import { useBackend, useLocalState } from '../backend';
+import { Autofocus, Box, Button, Section, Stack } from '../components';
+import { Window } from '../layouts';
+import { InputButtons } from './common/InputButtons';
+import { Loader } from './common/Loader';
+
+type KeyInputData = {
+  init_value: string;
+  large_buttons: boolean;
+  message: string;
+  timeout: number;
+  title: string;
+};
+
+const isStandardKey = (event): boolean => {
+  return (
+    event.key !== KEY.Alt &&
+    event.key !== KEY.Control &&
+    event.key !== KEY.Shift &&
+    event.key !== KEY.Escape
+  );
+};
+
+const KEY_CODE_TO_BYOND: Record<string, string> = {
+  DEL: 'Delete',
+  DOWN: 'South',
+  END: 'Southwest',
+  HOME: 'Northwest',
+  INSERT: 'Insert',
+  LEFT: 'West',
+  PAGEDOWN: 'Southeast',
+  PAGEUP: 'Northeast',
+  RIGHT: 'East',
+  SPACEBAR: 'Space',
+  UP: 'North',
+};
+
+const DOM_KEY_LOCATION_NUMPAD = 3;
+
+const formatKeyboardEvent = (event): string => {
+  let text = '';
+
+  if (event.altKey) {
+    text += 'Alt';
+  }
+
+  if (event.ctrlKey) {
+    text += 'Ctrl';
+  }
+
+  if (event.shiftKey && !(event.keyCode >= 48 && event.keyCode <= 57)) {
+    text += 'Shift';
```

Comment:
These should be displayed like `Shift+2` instead of `Shift2` imo. The rest of this looks pretty good

Reply:
https://github.com/ParadiseSS13/Paradise/assets/69762909/d06d54bf-2d48-4e63-89a5-a79024295c75

Reply:
Pain, thats fine then